packages/components/src/modifiers/hds-anchored-position.js

```diff
@@ -0,0 +1,223 @@
+import { modifier } from 'ember-modifier';
+import { assert } from '@ember/debug';
+
+import {
+  autoUpdate,
+  computePosition,
+  offset,
+  flip,
+  shift,
+  limitShift,
+  autoPlacement,
+  arrow,
+  // ---
+  // this could be used in the future if we want to give consumers an option to hide the "floating" element when the "anchor" hides from the viewport
+  // see: https://floating-ui.com/docs/hide
+  // hide,
+  // ---
+  // this could be used in the future if we want to give consumers an option to:
+  // - let the "floating" element auto-resize when there is not enough space (usually vertical) in the viewport to contain the entire "floating" element
+  // - let the "floating" element match the width of the "trigger" (it may have min/max width/heigh via CSS too)
+  // see: https://floating-ui.com/docs/size
+  // notice: below you can find a preliminary code implementation that was tested and worked relatively well
+  // size,
+  // ---
+} from '@floating-ui/dom';
+
+export const DEFAULT_PLACEMENT = 'bottom';
+export const PLACEMENTS = [
+  'top',
+  'top-start',
+  'top-end',
+  'right',
+  'right-start',
+  'right-end',
+  'bottom',
+  'bottom-start',
+  'bottom-end',
+  'left',
+  'left-start',
+  'left-end',
+];
+
+// we use this function to process all the options provided to the modifier in a single place,
+// in relation to the Floating UI APIs, and keep the modifier code more clean/simple
+export const getFloatingUIOptions = (options) => {
+  let {
+    placement = DEFAULT_PLACEMENT,
+    strategy = 'absolute', // we don't need to use `fixed` if we use the Popover API for the "floating" element (it puts the element in the `top-layer`)
+    offsetOptions,
+    flipOptions = { padding: 8 },
+    shiftOptions = { padding: 8, limiter: limitShift() },
+    autoPlacementOptions = { padding: 8 },
+    middlewareExtra = [],
+    enableCollisionDetection,
+    arrowOptions,
+  } = options;
+
+  // we build dynamically the list of middleware functions to invoke, depending on the options provided
+
+  const middleware = [];
+
+  // https://floating-ui.com/docs/offset
+  middleware.push(offset(offsetOptions));
+
+  // https://floating-ui.com/docs/flip
+  // https://floating-ui.com/docs/shift
+  // https://floating-ui.com/docs/autoPlacement
+  if (
+    enableCollisionDetection === true ||
+    enableCollisionDetection === 'flip'
+  ) {
+    middleware.push(flip(flipOptions));
+  }
+  if (
+    (enableCollisionDetection === true) |
```

Comment:
This shouldn't be `||` vs. `|` here as in the above if statement?

Reply:
I am not sure how I missed this! Great catch!